docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,307 @@
+
+### Custom Enchantment Effect Components
+
+Enchantment effect component types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`, which takes a `DataComponentType<?>`. For example, you could register an enchantment effect component that can store an `ExampleEffectData` object as follows:
+
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ExampleEffectData>>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<ExampleEffectData>builder()
+            .persistent(ExampleEffectData.CODEC)
+            .build());
+```
+
+There are no inheritance requirements on the data held by an enchantment effect component.
+
+### `ConditionalEffect`
+Wrapping the type in `ConditionalEffect<?>` allows the enchantment effect component to take effect based on conditions informed by a [LootContext]. 
+
+Specifically, each `ConditionalEffect` contains another effect component, along with an `Optional<LootItemCondition>`. Since `LootItemContext` is a `Predicate<LootContext>`, it can be tested against a specified `LootContext` using `LootItemContext#test`.
+
+`ConditionalEffect` wraps this behavior, allowing one to simply call `ConditionalEffect#matches(LootContext context)` to determine if the effect should be allowed to run.
+
+Vanilla adds an additional helper method to further streamline the process of checking these conditions: `Enchantment#applyEffects()`. This method takes a `List<ConditionalEffect<T>>`, evaluates the conditions, and runs a `Consumer<T>` on each `T` contained by a `ConditionalEffect` whose condition was met. Since many vanilla enchantment effect components are defined as a `List<ConditionalEffect<?>>`, these can be directly plugged into the helper method like so:
+
+```java
+// `enchant` is an Enchantment instance.
+// `lootContext` is a LootContext instance.
+Enchantment.applyEffects(
+    enchant.getEffects(EnchantmentEffectComponents.KNOCKBACK), // Or whichever other List<ConditionalEffect<T>> you want
+    lootContext,
+    (effectData) -> // Use the effectData (in this example, an EnchantmentValueEffect) however you want.
+);
+```
+
+Registering a custom `ConditionalEffect`-wrapped enchantment effect component type can be done as follows:
+
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ConditionalEffect<ExampleData>>> EXAMPLE_CONDITIONAL_EFFECT =
+    ENCHANTMENT_COMPONENT_TYPES.register("example_conditional",
+        () -> DataComponentType.ConditionalEffect<ExampleData>builder()
+            .persistent(ConditionalEffect.codec(Unit.CODEC, LootContextParamSets.EMPTY))
+            .build());
```

Comment:
The example doesn't line up... I honestly think you should just create a dummy object to reference. You don't need to implement the codec methods, but you should show that this is where it's coming from rather than referencing `ExampleData`, then passing in a `Unit#CODEC`, which is not `ExampleData`.

Reply:
I missed that I was still using the Unit codec, sorry.